modules/machineset-creating-azure-ultra-disk.adoc

```diff
@@ -0,0 +1,291 @@
+
+. Extract the disabling template value by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-machine-api get secret worker-user-data --template='{{index .data.disableTemplating | base64decode}}' | jq > disableTemplating.txt
+----
++
+where #need to understand what these values are and where they come from#
+
+. Merge the two files to create a datasecret file to be used by running the following command:
++
+[source,terminal]
+----
+oc -n openshift-machine-api create secret generic worker-user-data-x5 \
+--from-file=userData=userData.txt \
+--from-file=disableTemplating=disableTemplating.txt
+----
++
+where #need to understand what these values are and where they come from#
+endif::mapi[]
+
+. Edit or copy an existing Azure `MachineSet` custom resource (CR) by running the following command:
++
+[source,terminal]
+----
+$ oc edit machineset <machine-set-name>
+----
++
+where `<machine-set-name>` is the machine set that you want to provision machines on ultra disks.
+
+. Add the following lines in the positions indicated:
++
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+  ...
+spec:
+  ...
+  template:
+    ...
+    spec:
+      metadata:
+        ...
+        labels:
+          ...
+          disk: ultrassd <1>
+          ...
+      providerSpec:
+        value:
+          ...
+          ultraSSDCapability: Enabled <1>
+ifdef::mapi[]
+          dataDisks: <1>
+          - nameSuffix: ultrassd
+            lun: 0
+            diskSizeGB: 4
+            deletionPolicy: Delete
+            cachingType: None
+            managedDisk:
+              storageAccountType: UltraSSD_LRS
+endif::mapi[]
+             ...
+----
++
+<1> These lines enable the use of ultra disks.
+ifdef::mapi[For `dataDisks`, include the entire stanza.]
+
+. Create a machine set using the updated configuration by running the following command:
++
+[source,terminal]
+----
+$ oc create -f <machine-set-name>.yaml
+----
+
+ifdef::pvc[]
+. Create a storage class that contains the following YAML definition:
++
+[source,yaml]
+----
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  creationTimestamp: "2022-04-11T11:02:53Z"
+  name: ultra-disk-sc <1>
+  resourceVersion: "56877"
+  uid: 08b2d2ca-adbf-4f6e-b333-1612daf374ad
+parameters:
+  cachingMode: None
+  diskIopsReadWrite: "2000"
+  diskMbpsReadWrite: "320"
+  kind: managed
+  skuname: UltraSSD_LRS
+provisioner: disk.csi.azure.com <2>
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer <3>
+----
+<1> Specify the name of the storage class. This procedure uses `ultra-disk-sc` for this value.
+<2> For Azure Kubernetes Service (AKS) version 1.21 or later, use `disk.csi.azure.com`. For earlier versions of AKS, use `kubernetes.io/azure-disk`.
+<3> Optional: Specify this parameter to wait for the creation of the pod that will use the disk.
+#Which of the above lines are required? Which lines have to say something specific vs. being the user's choice?#
```

Comment:
line 170 , 172 and 173 are not needed , they are autogenerated.
176 and 177 are again choice administrator can use as per his requirement.